data-raw/nanostring.R

```diff
@@ -223,18 +214,18 @@ ampad_modules_fc %>%
 
 ns_control <- nanostring_with_metadata %>%
   filter(mouse_model == "C57BL6J") %>%
-  select(gene, specimen_id, value, sex, age_group, mouse_model) %>%
+  select(gene, specimen_id, value, sex, age, mouse_model) %>%
   pivot_wider(names_from = specimen_id, values_from = value, names_prefix = "value_") %>%
-  group_by(sex, age_group, mouse_model) %>%
+  group_by(sex, age, mouse_model) %>%
   nest() %>%
   ungroup() %>%
```

Comment:
Just making sure you want this ungroup here.  As far as I understand this, the ungroup will replicate the nested values across the rows is opens up from the groups, right?

Reply:
The ungroup only removes the "grouping" attribute from the data set, it doesn't actually affect any of the values otherwise.

Here's how it looks after the `nest()`:

```r
# A tibble: 6 x 4
# Groups:   sex, age, mouse_model [6]
  sex      age mouse_model data               
  <chr>  <dbl> <chr>       <list>             
1 Female    12 C57BL6J     <tibble [765 × 33]>
2 Male       8 C57BL6J     <tibble [765 × 33]>
3 Male      12 C57BL6J     <tibble [765 × 33]>
4 Male       4 C57BL6J     <tibble [765 × 33]>
5 Female     8 C57BL6J     <tibble [765 × 33]>
6 Female     4 C57BL6J     <tibble [765 × 33]>
```

And then after `ungroup()`

```r
# A tibble: 6 x 4
  sex      age mouse_model data               
  <chr>  <dbl> <chr>       <list>             
1 Female    12 C57BL6J     <tibble [765 × 33]>
2 Male       8 C57BL6J     <tibble [765 × 33]>
3 Male      12 C57BL6J     <tibble [765 × 33]>
4 Male       4 C57BL6J     <tibble [765 × 33]>
5 Female     8 C57BL6J     <tibble [765 × 33]>
6 Female     4 C57BL6J     <tibble [765 × 33]>
```

An alternate way to do this is to use `group_nest()`, i.e. `group_nest(sex, age, mouse_model`) instead of the `group_by() %>% nest() %>% ungroup()`:

```r
# A tibble: 6 x 4
  sex      age mouse_model                data
  <chr>  <dbl> <chr>       <list<tbl_df[,33]>>
1 Female     4 C57BL6J              [765 × 33]
2 Female     8 C57BL6J              [765 × 33]
3 Female    12 C57BL6J              [765 × 33]
4 Male       4 C57BL6J              [765 × 33]
5 Male       8 C57BL6J              [765 × 33]
6 Male      12 C57BL6J              [765 × 33]
```

Not sure if that answers your Q! My goal is just to get a nested data frame for each sex/age combo, so that I can join the variant and control dfs and run the model on this data:

```r
   mouse_model       sex      age data_variant        data_control      
   <chr>             <chr>  <dbl> <list>              <list>            
 1 LOAD1             Male      12 <tibble [765 × 29]> <tibble [765 × 7]>
 2 LOAD1             Female    12 <tibble [765 × 24]> <tibble [765 × 6]>
 3 LOAD1.MthfrC677T  Male      12 <tibble [765 × 12]> <tibble [765 × 7]>
 4 LOAD1.MthfrC677T  Female    12 <tibble [765 × 15]> <tibble [765 × 6]>
 5 LOAD1.Clasp2L163P Female     4 <tibble [765 × 7]>  <tibble [765 × 7]>
 6 LOAD1.Clasp2L163P Male       4 <tibble [765 × 7]>  <tibble [765 × 7]>
 7 LOAD1.Snx1D465N   Male       4 <tibble [765 × 7]>  <tibble [765 × 7]>
 8 LOAD1             Male       4 <tibble [765 × 14]> <tibble [765 × 7]>
 9 LOAD1             Female     4 <tibble [765 × 17]> <tibble [765 × 7]>
10 LOAD1.Snx1D465N   Female     4 <tibble [765 × 7]>  <tibble [765 × 7]>
# … with 50 more rows
```

Reply:
I actually just took a screen shot of this.  Thanks for the detailed explanation.  That does answer it.  I thought ungrouping here would "explode" the groups into repeated rows for mouse_model, sex, and age, while preserving the value for data for each of them.  In other words you'd end up with an N:1 relationship between the groups and the column labeled data.  And that's not the case, and what you did does accomplish what you wanted!  Thanks again for explaining it! (: